### PR TITLE
Throws more explicit error message for issue #14

### LINKED ISSUE #14: NullReferenceException when loading project file with missing AssemblyName/OutputType
I work some VB/C# projects which (for whatever reason) don't have `AssemblyName` or `OutputType` elements in their project XML. Visual Studio loads and msbuild builds these projects perfectly well, but attempting to load a solution containing them via nunit3-console throws a `NullReferenceException`. I didn't realize what the problem was until I pulled down the vs-project-loader code and looked at the relevant methods in the stack trace (namely, the `assemblyName.InnerText` and `outputTypeElement.InnerText` lines in `VSProject.cs:LoadMSBuildProject`).

I'm fine if vs-project-loader errors out when attempting to load these project files, but it would be nice if I didn't have to dig into the code to see what exactly was wrong with the file. Willing to submit a PR myself to throw a more specific exception. Thanks!

### PR DESCRIPTION
**Pull request by mrshanahan**
Fixes #14

### CODE REVIEW

**Comment by ChrisMaddock** [comment]
Looks good, thanks!

**Comment by ChrisMaddock** [comment]
Merged - thanks for the contribution!

**Comment by mrshanahan** [comment]
Great! Thanks for the quick response!

File: src/extension/VSProject.cs [hunk 1]
```diff
@@ -348,9 +348,13 @@ private void LoadMSBuildProject()
             MsBuildDocument = _doc;
 
             XmlElement assemblyNameElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:AssemblyName", namespaceManager);
+            if (assemblyNameElement == null)
+                throw new ApplicationException("Missing required element AssemblyName");
             string assemblyName = assemblyNameElement.InnerText;
 
             XmlElement outputTypeElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:OutputType", namespaceManager);
+            if (outputTypeElement == null)
+                throw new ApplicationException("Missing required element OutputType");
```
Comment (CharliePoole) [inline src/extension/VSProject.cs]: We generally avoid direct use of ApplicationException. In the engine and extensions, we use NUnitEngineException, with an appropriate message.

Comment (ChrisMaddock) [inline src/extension/VSProject.cs]: I wondered why ApplicationException - but it follows convention in the rest of this file. This exception is also caught and converted to an Argument exception in the next file up.

Comment (mrshanahan) [inline src/extension/VSProject.cs]: I was attempting to follow the style that were being used for legacy MSBuild projects in the same file, which were throwing an ApplicationException with a similar message if the OutputType/AssemblyName attributes were missing. I can certainly change to use if you feel it warrants it though.

Comment (mrshanahan) [inline src/extension/VSProject.cs]: And the exception is a little ugly - a raw stack trace in the console. I haven't tried it in Visual Studio yet, but I'll update here.

**Comment by CharliePoole** [comment]
@mrshanahan I realize now that my comment about the VS adapter is a red herring! ☹️ We don't use this extension in VS because VS itself gives us the assemblies.

I didn't look at the rest of the file - a drawback of reviewing just the changes in GitHub I guess. I suppose this all comes from code written for NUnit V2 and simply moved here with a new interface stuck on it. We did add NUnitEngineException to the API assembly so it could be used by extensions, but I'm OK with a separate issue to root out the uses of ApplicationException.

The big issue is whether we should be doing more than just throwing an exception. Since this is merged, let's continue the discussion on #14.

**Comment by mrshanahan** [comment]
Sounds good! Thanks you!

**Pull request merged**